### PR TITLE
Load synonyms from system index for analyzers

### PR DESCRIPTION
**Pull request by mayya-sharipova**
- Create a new option of "synonyms_set" for synonym set filter that specifies which synonyms set to be loaded from the system ".synonyms" index
- On index creation for this option load synonyms set  from index
- If synonyms set doesn't exist, index creation request still succeeds, but shards are not allocated, so the cluster state will be read.

Note: this is a temporary solution, as:
- No check is done on master node, as fake synonyms are provided
- On shard on index creation we use a blocking operation in the cluster applier thread

### CODE REVIEW

**Review by javanna — APPROVED** [review]
LGTM, this allows us to make progress under a feature flag while discussing on a long term solutions for loading synonyms on index creation.

File: modules/analysis-common/src/main/java/org/elasticsearch/analysis/common/SynonymTokenFilterFactory.java [hunk 1]
```diff
@@ -138,18 +154,54 @@ SynonymMap buildSynonyms(Analyzer analyzer, ReaderWithOrigin rules) {
...
+
         if (settings.getAsList("synonyms", null) != null) {
             List<String> rulesList = Analysis.getWordList(env, settings, "synonyms");
             StringBuilder sb = new StringBuilder();
             for (String line : rulesList) {
                 sb.append(line).append(System.lineSeparator());
             }
             return new ReaderWithOrigin(new StringReader(sb.toString()), "'" + name() + "' analyzer settings");
+        } else if ((settings.get("synonyms_set") != null) && SynonymsAPI.isEnabled()) {
+            if (analysisMode != AnalysisMode.SEARCH_TIME) {
```
Comment (carlosdelest) [inline modules/analysis-common/src/main/java/org/elasticsearch/analysis/common/SynonymTokenFilterFactory.java:166]: Maybe we can refactor this into a separate method for retrieving the `ReaderWithOrigin` and make clearer the case distinction for the different synonyms origin.

**Review by carlosdelest — APPROVED** [review]
LGTM 👍

File: modules/analysis-common/src/main/java/org/elasticsearch/analysis/common/SynonymTokenFilterFactory.java (hunk 1 above)
Comment (mayya-sharipova) [reply modules/analysis-common/src/main/java/org/elasticsearch/analysis/common/SynonymTokenFilterFactory.java:163]: Thanks for the feedback, addressed in 334caf6cc4ff8919a797b4688a28b5bbd824e945

**Pull request merged**